specs/proposals/blob-signing.md

```diff
@@ -0,0 +1,335 @@
+
+2. Publish both the file and corresponding signature using any file transfering mechanism. Both the file and its signature can be packaged into one file (e.g., a tarball) for transferring.
+
+3. Download or fetch both the file and signature. If they are packaged into one file, for example, a tarball, unpackage the file to get separate file and signature.
+
+4. Verify the file against the signature:
+
+   - Add the root CA certificate and set up the trust store (The same experience for verifying container images):
+
+        ```shell
+        notation cert add --type ca --store myCACerts root.crt
+        ```
+
+     Confirm the certificate is added.
+
+        ```shell
+        notation cert ls
+        ```
+
+        > [!NOTE]
+        > Learn more options using `notation cert ls --help`.
+
+    - Set up trust policy for blobs with a new command `notation blob policy init`. This command streamlines the process, eliminating the need for users to consult documentation for the correct trust policy format and preventing the accidental use of policies intended for OCI artifact verification.
+
+        ```shell
+        notation blob policy init --name "myBlobPolicy" --trust-store "ca:myCACerts" --trust-identity "x509.subject:C=US,ST=WA,O=wabbit-network.io"
+        ```
+
+        Show the policies configured for verifying blobs:
+
+        ```shell
+        notation blob policy show
+        ```
+
+        > [!NOTE]
+        > See the [section](#manage-blob-policies) for more commands for `notation blob policy`.
+
+    - Verify the signature:
+
+        ```shell
+        notation blob verify --policy-name myBlobPolicy --signature sbom.json.cose.sig sbom.json
+        ```
+
+        If a [global policy](https://github.com/notaryproject/specifications/blob/main/specs/trust-store-trust-policy.md#blob-trust-policy) is set, you can skip the `--policy-name` flag.
+
+### Blob Sign and verify with registry-based distribution
+
+For registry-based distribution, such as using an OCI-compliant container registry, the process is similar but includes additional steps for pushing blobs and sigantures to the registry.
+
+**Prerequisites:**
+- SBOMs, release artifacts or other files are created and ready for publishing.
+
+**Steps:**
+
+1. Sign a file on filesystem with a specified signature envelope:
+
+    ```shell
+    notation blob sign --id myKeyId --signature-format "cose" --media-type "application/spdx+json" sbom.json
+    ```
+
+    This generates a signature named `sbom.json.cose.sig` in the current working directory.
+
+2. Publish the file and signature to the OCI-compliant registry:
+
+    Use [ORAS tool](https://oras.land/) to push the blob file:
+
+    ```shell
+    oras push --artifact-type "application/spdx+json" sbom.json:application/json myregistry/myapp/sbom:v1
```

Comment:
Can I also use `oras attach`?

Reply:
This command pushes the SBOM file, while `oras attach` is used later to attach the signature.

In this scenario, the SBOM is generated for a software application, not for a container image. Therefore, the example uses `oras push`, not to attach the SBOM to a container image in the registry.